src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+
+  for (let index = 0; index < actions.length; index++) {
+    const action = actions[index];
+    let newState;
+    const prevState = states[index - 1] || state;
+
+    if (action.type === 'addProperties') {
+      newState = {
+        ...prevState,
+        ...action.extraData,
+      };
+    }
+
+    if (action.type === 'removeProperties') {
+      newState = { ...prevState };
```

Comment:
you should copy object before all conditions, that way you wouldn't repeat yourself